src/pages/index.js

```diff
@@ -76,21 +90,113 @@ const popupWithImage = new PopupWithImage("#image-popup");
 popupWithImage.setEventListeners();
 
 //Section
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      renderCard(cardData);
-    },
-  },
-  "cards__list"
-);
+let section;
+api
+  .getInitialCards()
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (cardData) => {
+          const cardEl = renderCard(cardData);
+          section.addItem(cardEl);
+        },
+      },
+      "cards__list"
+    );
+    section.renderItems();
+  })
+
+  .catch((error) => {
+    console.error(error);
+  });
+
+api
+.getUserInfo()
+  .then((user) => {
+    userInfo.setUserInfo(user.name, user.about);
+    userInfo.setAvatar(user.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
-section.renderItems();
 //UserInfo
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo(".profile__title", ".profile__description", ".profile__image");
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.name, inputValues.description);
-  editPopup.close();
+  editPopup.setLoading(true);
+  api.editUserInfo(inputValues).then(() => {
+    userInfo.setUserInfo(inputValues.name, inputValues.description);
+    editPopup.close();
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => {
+    editPopup.setLoading(false);
+  });
+}
+
+//EditAvatar
+
+const updateAvatarForm = new PopupWithForm("#profile-image-modal", handleAvatarSubmit);
+function handleAvatarSubmit(inputValues) {
+  updateAvatarForm.setLoading(true);
+  api.updateAvatar(inputValues.link).then((user) => {
+    userInfo.setAvatar(user.avatar);
+    updateAvatarForm.setLoading(false);
+  });
+  updateAvatarForm.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

**Please, correct it everywhere**